src/conn/conn_tiered.c

```diff
@@ -194,8 +197,21 @@ __wt_tier_do_flush(
     WT_RET(storage_source->ss_flush(
       storage_source, &session->iface, bucket_fs, local_name, obj_name, NULL));
 
-    WT_WITH_CHECKPOINT_LOCK(session,
-      WT_WITH_SCHEMA_LOCK(session, ret = __tier_flush_meta(session, tiered, local_uri, obj_uri)));
+    /*
+     * Flushing the metadata grabs the data handle with exclusive access, and the data handle may be
+     * held by the thread that queues the flush tier work item. As a result, the handle may be busy,
+     * so retry as needed, up to a few seconds.
+     */
+    for (tries = 1, millis = 100; tries <= 6; tries++) {
+        WT_WITH_CHECKPOINT_LOCK(session,
+          WT_WITH_SCHEMA_LOCK(
+            session, ret = __tier_flush_meta(session, tiered, local_uri, obj_uri)));
+        if (ret != EBUSY)
+            break;
+        __wt_sleep(0, millis * WT_THOUSAND);
+        millis *= 2;
+        /* TODO: tiered: statistic for retries */
```

Comment:
I suggest adding this statistic. It is two additional lines, the definition in the `stat_data.py` file and one increment here. That doesn't seem worth a new ticket.

Reply:
Done.